crates/neon/src/macros.rs

```diff
@@ -122,6 +124,61 @@ pub use neon_macros::main;
 /// }
 /// ```
 ///
+/// ### Async Functions
+///
+/// The [`export`] macro can export `async fn`, converting to a JavaScript `Promise`, if a global
+/// future executor is registered. See [`neon::set_global_executor`](crate::set_global_executor) for
+/// more details.
+///
+/// ```
+/// # #[cfg(all(feature = "napi-6", feature = "futures"))]
+/// # {
+/// #[neon::export]
+/// async fn add(a: f64, b: f64) -> f64 {
+///     a + b
+/// }
+/// # }
+/// ```
+///
+/// A `fn` that returns a [`Future`](std::future::Future) can be annotated with `#[neon::export(async)]`
+/// if it needs to perform some setup on the JavaScript main thread before running asynchronously.
+///
+/// ```
+/// # #[cfg(all(feature = "napi-6", feature = "futures"))]
+/// # {
+/// # use std::future::Future;
+/// # use neon::prelude::*;
+/// #[neon::export(async)]
+/// fn add(_cx: &mut FunctionContext, a: f64, b: f64) -> NeonResult<impl Future<Output = f64>> {
```

Comment:
@dherman This doesn't need the unused `_cx`, but I was trying to show that it _can_ use a `cx`. A few options:
* Leave it as is
* Switch to `console.log` to demonstrate using it
* Remove it
* Comment it out to show that it _could_ be used but isn't required

```suggestion
/// fn add(/* cx: &mut FunctionContext, */ a: f64, b: f64) -> NeonResult<impl Future<Output = f64>> {
```

What do you think?

Reply:
Honestly I feel like we should have a shorthand for the console, so you could do:

```rust
cx.console()?.log(&mut cx, "Hello from the main JavaScript thread!")?;
```

But for now why don't we just remove the `cx` argument entirely and the `NeonResult`, and we can add it back after we merge #1056, because then we could at least write:

```rust
    cx.global("console")?
        .get(&mut cx, "log")?
        .bind(&mut cx)
        .arg("Hello from the main JavaScript thread!")
        .apply()?;
```

Reply:
Actually I think it's probably best to leave them out for good. The docs don't talk about taking a `FunctionContext` until the bottom of the page, so it's best not to jump ahead. IMO we should keep it simple with `println!` for these examples.